binding-hs/src/Client.hs

```diff
@@ -0,0 +1,192 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Client
+  ( envCreate
+  , envListAll
+  , envReset
+  , envStep
+  , envActionSpaceInfo
+  , envActionSpaceSample
+  , envActionSpaceContains
+  , envObservationSpaceInfo
+  , envMonitorStart
+  , envMonitorClose
+  , envClose
+  , upload
+  , shutdownServer
+  , EnvID (..)
+  , InstID (..)
+  , Environment (..)
+  , Observation (..)
+  , Step (..)
+  , Outcome (..)
+  , Info (..)
+  , Action (..)
+  , Monitor (..)
+  , Config (..)
+  ) where
+
+import           Data.Aeson          (FromJSON, Object (..), ToJSON)
+import qualified Data.HashMap.Strict as HM (HashMap (..))
+import           Data.Proxy          (Proxy (..))
+import qualified Data.Text           as T (Text)
+import           GHC.Generics
+import           Network.HTTP.Client (Manager, defaultManagerSettings,
+                                      newManager)
+import           Servant.API
+import           Servant.Client      (BaseUrl (..), ClientM, client)
+import           Servant.HTML.Lucid  (HTML)
+
+--------------------------------------------------------------------------------
+-- | Data types and instances
+
+newtype EnvID = EnvID { env_id :: T.Text }
```

Comment:
I don't think Text is the right type for an environment id. Environments are relatively static and need very little maintanence, deserving their own type. Unfortunately, openai environments are inconsistently named so writing a new sum type also means writing out the show instance. It kind of sucks, but I'm of the opinion that the right thing to do is to have [something like this](https://github.com/openai/gym-http-api/pull/26/files#diff-e8341c1b62902a3181916d379cba4969), seed it with some of your favorite environments, then let people submit PRs if they want new environments to play with.